benchmarks/infill_density/infill_ascii_data.cc

```diff
@@ -0,0 +1,253 @@
+      return traction;
+    }
+
+
+    template <int dim>
+    void
+    InfillAsciiData<dim>::update()
+    {
+      Interface<dim>::update ();
+      Utilities::AsciiDataBoundary<dim>::update();
+    }
+
+
+    template <int dim>
+    void
+    InfillAsciiData<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Boundary traction model");
+      {
+        Utilities::AsciiDataBoundary<dim>::declare_parameters(prm,
+                                                              "$ASPECT_SOURCE_DIR/tests/infill_density/",
+                                                              "triangle_load.txt");
+        prm.enter_subsection("Infill ascii");
+        {
+          prm.declare_entry ("Rock density", "2800",
+                             Patterns::List(Patterns::Double(0.)),
+                             "Density of the volcanic edifice that infills the flexural moat.");
+          prm.declare_entry ("Sediment density", "2300",
+                             Patterns::List(Patterns::Double(0.)),
+                             "Density of the sediment that infills the flexural moat.");
+          prm.declare_entry ("Height for specifying rock infill", "500",
+                             Patterns::List(Patterns::Double(0.)),
+                             "If the load defined in the ASCII file has a height equal to or greater than "
+                             "the Height for specifying rock infill, then the infill material will be rock, "
+                             "otherwise it will be sediment");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    InfillAsciiData<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Boundary traction model");
+      {
+        Utilities::AsciiDataBoundary<dim>::parse_parameters(prm);
+        prm.enter_subsection("Infill ascii");
+        {
+          rock_density = prm.get_double("Rock density");
+          sediment_density = prm.get_double("Sediment density");
+          rock_infill_height = prm.get_double("Height for specifying rock infill");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace BoundaryTraction
+  {
+    ASPECT_REGISTER_BOUNDARY_TRACTION_MODEL(InfillAsciiData,
+                                            "infill ascii data",
+                                            "Implementation of a model in which the boundary "
+                                            "traction is derived from files containing pressure data "
+                                            "in ascii format. The pressure given in the data file is "
+                                            "applied as traction normal to the surface of a given boundary, "
+                                            "pointing inwards. Note the required format of the "
```

Comment:
This looks very much like a description of a general ascii data boundary traction model. Can you add more details on what this specific mode does?